lib/IRGen/GenDistributed.cpp

```diff
@@ -113,6 +150,11 @@ static CanSILFunctionType getAccessorType(IRGenModule &IGM,
                             ParameterConvention::Direct_Guaranteed);
   };
 
+  auto getUIntParameter = [&]() {
+    return SILParameterInfo(Context.getUIntType()->getCanonicalType(),
+                            ParameterConvention::Direct_Guaranteed);
```

Comment:
Technically, `Direct_Unowned`?

Reply:
I picked guaranteed because the comment said that `caller guarantees its validity for the entirety of the call.` but I can change it to `Direct_Unowned` if that's better since I'm not privy to all of the nuances that differentiate the two.

Reply:
We use `Unowned` for trivial types.

```
  /// This argument is passed directly.  Its type may be trivial, or it may     
  /// simply be that the callee is not responsible for destroying it.  Its      
  /// validity is guaranteed only at the instant the call begins.               
  Direct_Unowned,                                                               
                                                                                
  /// This argument is passed directly.  Its type is non-trivial, and the caller
  /// guarantees its validity for the entirety of the call.                     
  Direct_Guaranteed,   
  ```

Reply:
So since all these are either pointers or i{{64|32}} I should have used `Unowned` but if I want to e.g. pass a reference to class or struct it should be `Guaranteed`?

Reply:
Updated!